spec/support/span_helpers.rb

```diff
@@ -96,4 +96,13 @@ def description_of(actual)
       end
     end
   end
+
+  RSpec::Matchers.define :a_span_with do |expected|
+    match do |actual|
+      actual.instance_of?(Datadog::Tracing::Span) &&
+        expected.all? do |key, value|
+          actual.__send__(key) == value
+        end
+    end
+  end
```

Comment:
Isn't this the same as the built-in matcher [have_attributes](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/have-attributes-matcher)?

Reply:
It does [all?](https://github.com/rspec/rspec-expectations/blob/1bc30dc41665ade783340d71e1d8f30bf3c9b995/lib/rspec/matchers/built_in/have_attributes.rb#L29) and [__send__](https://github.com/rspec/rspec-expectations/blob/1bc30dc41665ade783340d71e1d8f30bf3c9b995/lib/rspec/matchers/built_in/have_attributes.rb#L80) just like this matcher.

Reply:
Yeah, it is more or less 
`be_an_instance_of(Datadog::Tracing::Span).and(have_attributes(name: 'root', service_entry: true))` ,
but `have_attributes` would not work for private methods.